sympy/physics/units/dimensions.py

```diff
@@ -206,8 +207,6 @@ def _get_dimensional_dependencies_for_name(name):
             return {k: v for (k, v) in ret.items() if v != 0}
 
         if name.is_Pow:
-            if name.exp == 0:
-                return {}
```

Comment:
There might be expressions in dimensions, which I think should be exposed to the multiplication by zero.